src/accelerate/utils.py

```diff
@@ -350,6 +350,39 @@ def gather(tensor):
         return tensor
 
 
+def _gpu_gather_object(object: Any):
+    def _gpu_gather_object_one(object: Any):
+        output_objects = [None for _ in range(state.num_processes)]
```

Comment:
```suggestion
        output_objects = [None for _ in range(AcceleratorState().num_processes)]
```
`state` is not defined.